### ISSUE TITLE
Timezone is not updated during summer saving

### ISSUE THREAD

**Issue by ibahnasy** [nextcloud/calendar#5468]
### Steps to reproduce

1. Set Calendar timezone to Africa/Cairo
2. Create an event and sync it on another device that has the same timezone.

### Expected behavior

Event on NextCloud Calendar and the other device must have the same time.

### Actual behaviour

There is 1 hour difference.
Egypt/Cairo is GMT +2 but in summer it's +3 but Calendar app is only using GMT +2 all time.

### Calendar app version

4.4.5

### CalDAV-clients used

DAVx5

### Browser

Firefox 102.15 ESR

### Client operating system

Debian 12

### Server operating system

Ubuntu 22.04

### Web server

Nginx

### Database engine version

MariaDB

### PHP engine version

PHP 8.1

### Nextcloud version

27.0.2

### Updated from an older installed version or fresh install

Fresh install

**Comment by ChristophWurst** [comment]
Our timezone info might be out of data (again), like with https://github.com/nextcloud/calendar/issues/5234.

Did Cairo summer time change recently?

**Comment by ChristophWurst** [comment]
The action didn't catch any updates https://github.com/nextcloud/calendar-js/actions/workflows/update_timezones.yml

**Comment by ibahnasy** [comment]
Yes summer time was not used for many years but then they started using it recently.
------- Original Message -------

**Comment by ibahnasy** [comment]
Currently we are on EEST (UTC+3) but Cairo is basically UTC+2.
------- Original Message -------

**Comment by JohannesGGE** [comment]
Interesting bug.
Our timezone data is up to date I think. [zones.json](https://github.com/nextcloud/calendar-js/blob/4b17c69239c24faf288f97562cc1cf27ec428cc2/resources/timezones/zones.json#L479-L483) :
```
"Africa/Cairo": {
      "ics": [
        "BEGIN:DAYLIGHT\r\nTZOFFSETFROM:+0200\r\nTZOFFSETTO:+0300\r\nTZNAME:EEST\r\nDTSTART:19700424T000000\r\nRRULE:FREQ=YEARLY;BYMONTH=4;BYDAY=-1FR\r\nEND:DAYLIGHT",
        "BEGIN:STANDARD\r\nTZOFFSETFROM:+0300\r\nTZOFFSETTO:+0200\r\nTZNAME:EET\r\nDTSTART:19701030T000000\r\nRRULE:FREQ=YEARLY;BYMONTH=10;BYDAY=-1FR\r\nEND:STANDARD"
      ],
      "latitude": "+0300300",
      "longitude": "+0311500"
    },
```

The Nextcloud Calendar App display correct as well:

|Berlin UTC+2|Cairo UTC+3|
|--|--|
|![image](https://github.com/nextcloud/calendar/assets/74607597/7d89b40d-bb72-432c-b4e1-7634f23d6e3e)|![image](https://github.com/nextcloud/calendar/assets/74607597/0549ceaa-763a-46e8-aee4-734d7af4687f)|

But I can confirm, that Events created with `Africa/Cairo` displayed as UTC+2 in mobile Apps (tested Etar, Simple Calendar and Google Calendar App).

I checked the saved data and compared it with a Event created with `Europe/Kiev`, because they have the same timezone pattern as `Africa/Cairo` and work in external apps. 

```diff
BEGIN:VCALENDAR
CALSCALE:GREGORIAN
VERSION:2.0
PRODID:-//IDN nextcloud.com//Calendar app 4.5.1//EN
BEGIN:VEVENT
CREATED:20230926T095426Z
DTSTAMP:20230926T095611Z
LAST-MODIFIED:20230926T095611Z
SEQUENCE:3
-SEQUENCE:2
UID:c37e4ec2-ecb8-4b12-bd16-331206d612d0
DTSTART;TZID=Europe/Kiev:20230926T140000
DTEND;TZID=Europe/Kiev:20230926T150000
-DTSTART;TZID=Africa/Cairo:20230926T140000
-DTEND;TZID=Africa/Cairo:20230926T150000
STATUS:CONFIRMED
SUMMARY:14:00-15:00 Kiev
END:VEVENT
BEGIN:VTIMEZONE
TZID:Europe/Kiev
-TZID:Africa/Cairo
BEGIN:STANDARD
TZOFFSETFROM:+0300
TZOFFSETTO:+0200
TZNAME:EET
DTSTART:19701025T040000
-DTSTART:19701030T000000
RRULE:FREQ=YEARLY;BYMONTH=10;BYDAY=-1SU
-RRULE:FREQ=YEARLY;BYMONTH=10;BYDAY=-1FR
END:STANDARD
BEGIN:DAYLIGHT
TZOFFSETFROM:+0200
TZOFFSETTO:+0300
TZNAME:EEST
DTSTART:19700329T030000
-DTSTART:19700424T000000
RRULE:FREQ=YEARLY;BYMONTH=3;BYDAY=-1SU
-RRULE:FREQ=YEARLY;BYMONTH=4;BYDAY=-1FR
END:DAYLIGHT
END:VTIMEZONE
END:VCALENDAR
```
The marked lines (-) are the Cairo lines which differ from the Kiev data.

It seems that only the `DTSTART;TZID=Africa/Cairo:20230926T140000` and `DTEND;TZID=Africa/Cairo:20230926T150000` affect the display position in the Apps. If I change `Africa/Cairo` to `Europe/Kiev` in these two lines the event is displayed correct in the Apps. 

I'm confused :laughing:

**Comment by JohannesGGE** [comment]
I assume this is not a bug on our side. If you create an event in Etar or Google and try to set the timezone to Egypt there is just an Eastern European Standard Time (UTC+2) option. For Ukraine there is the Eastern European Summer Time option. So the Apps likely have old data.

**Comment by b3niup** [comment]
I have same issue. 
My timezone is Europe/Warsaw.

After change to winter time all events on mobile devices (Android via davx5 and iOS) are now shown with an 1 hour offset. 
Calendar in web app shows good time though.

I'm using nextcloud and calendar for few years now and it never happened before.

Nextcloud 30.0.1
Calendar 5.0.1

**Comment by st3iny** [comment]
I can't reproduce this any longer on an Android phone. The time zone data seems to have been updated in the meantime.

Please reopen the ticket if this is not the case.
